src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java

```diff
@@ -288,4 +328,49 @@ private void validateMultiTableProperty(SwitchId switchId, List<String> errorMes
             }
         }
     }
+
+    /**
+     * Check for equals endpoints.
+     *
+     * @param firstFlow a first flow.
+     * @param secondFlow a second flow.
+     */
+    @VisibleForTesting
+    void checkForEqualsEndpoints(RequestedFlow firstFlow, RequestedFlow secondFlow) throws InvalidFlowException {
+        List<Endpoint> endpoints = new ArrayList<>();
+        endpoints.add(new Endpoint(firstFlow.getSrcSwitch(),
+                firstFlow.getSrcPort(), firstFlow.getSrcVlan()));
+        endpoints.add(new Endpoint(firstFlow.getDestSwitch(),
+                firstFlow.getDestPort(), firstFlow.getDestVlan()));
+        endpoints.add(new Endpoint(secondFlow.getSrcSwitch(),
+                secondFlow.getSrcPort(), secondFlow.getSrcVlan()));
+        endpoints.add(new Endpoint(secondFlow.getDestSwitch(),
+                secondFlow.getDestPort(), secondFlow.getDestVlan()));
+
+        Set<Endpoint> checkSet = new HashSet<>();
+        for (Endpoint endpoint : endpoints) {
+            if (!checkSet.contains(endpoint)) {
+                checkSet.add(endpoint);
+            } else {
```

Comment:
need more details on that validation logic, please leave a comment

Reply:
Fixed